soc/xtensa/intel_adsp/ace/power.c

```diff
@@ -165,7 +171,7 @@ void power_gate_entry(uint32_t core_id)
 
 	soc_cpus_active[core_id] = false;
 	sys_cache_data_flush_range(soc_cpus_active, sizeof(soc_cpus_active));
-	k_cpu_idle();
+	wait_for_interrupt(0);
```

Comment:
This is not needed if you always do `waiti 0`. Just call `k_cpu_idle()` that it will end up calling your `arch_cpu_idle()` function. All state conditions could be checked there then ...